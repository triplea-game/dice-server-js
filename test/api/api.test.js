const mockNconf = {
  get: jest.fn(string => string),
};
const mockRollDice = jest.fn();
const mockValidator = jest.fn();
const mockEmailManager = jest.fn();
const mockDbHandler = jest.fn();
mockDbHandler.prototype.setupDb = jest.fn();


jest.mock('nconf', () => mockNconf);
jest.mock('../../src/api/dice-roller', () => ({
  roll: mockRollDice,
}));
jest.mock('../../src/api/validator', () => mockValidator);
jest.mock('../../src/api/email-manager.js', () => mockEmailManager);
jest.mock('../../src/api/db-handler', () => mockDbHandler);

const api = require('../../src/api/api');

const mockRouter = {
  get: jest.fn(),
  post: jest.fn(),
  use: jest.fn(),
};

describe('The API', () => {
  it('should setup the routes correctly', () => {
    const router = api(mockRouter, {
      name: 'Dummy DB',
    });

    expect(router).toBe(mockRouter);

    expect(mockNconf.get).toMatchSnapshot();
    expect(mockRollDice).toMatchSnapshot();
    expect(mockValidator).toMatchSnapshot();
    expect(mockEmailManager).toMatchSnapshot();
    expect(mockDbHandler).toMatchSnapshot();
    expect(mockDbHandler.prototype.setupDb).toMatchSnapshot();
    expect(mockRouter.get).toMatchSnapshot();
    expect(mockRouter.post).toMatchSnapshot();
    expect(mockRouter.use).toMatchSnapshot();
  });
});
