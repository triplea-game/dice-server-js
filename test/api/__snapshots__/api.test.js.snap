// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The API should setup the routes correctly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "email:smtp",
    ],
    Array [
      "email:display:server",
    ],
    Array [
      "email:display:sender",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": "email:smtp",
    },
    Object {
      "isThrow": false,
      "value": "email:display:server",
    },
    Object {
      "isThrow": false,
      "value": "email:display:sender",
    },
  ],
}
`;

exports[`The API should setup the routes correctly 2`] = `[MockFunction]`;

exports[`The API should setup the routes correctly 3`] = `
[MockFunction] {
  "calls": Array [
    Array [],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`The API should setup the routes correctly 4`] = `
[MockFunction] {
  "calls": Array [
    Array [
      mockConstructor {},
      "email:smtp",
      "email:display:server",
      "email:display:sender",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`The API should setup the routes correctly 5`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "name": "Dummy DB",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`The API should setup the routes correctly 6`] = `
[MockFunction] {
  "calls": Array [
    Array [],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`The API should setup the routes correctly 7`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/verify/:token",
      [Function],
      [Function],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`The API should setup the routes correctly 8`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/roll",
      [Function],
      [Function],
      [Function],
    ],
    Array [
      "/register",
      [Function],
      [Function],
    ],
    Array [
      "/register/:token",
      [Function],
    ],
    Array [
      "/unregister",
      [Function],
      [Function],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`The API should setup the routes correctly 9`] = `
[MockFunction] {
  "calls": Array [
    Array [
      [Function],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
